src/twister2/report/test_results_plugin.py

```diff
@@ -246,8 +246,8 @@ def pytest_configure(config: pytest.Config):
         return
 
     test_results_writers = []
-    if test_result_json_path := config.getoption('results_json_path'):
-        test_results_writers.append(JsonResultsReport(test_result_json_path))
+    test_result_json_path = os.path.join(config.getoption('output_dir'), config.getoption('results_json_path'))
```

Comment:
what if you use a custom result path, like `--result-json=/tmp/log/testresults.json` ?
If one want have a custom result file, then skip creating new path